libs/hpc-data/src/lib/flows.ts

```diff
@@ -0,0 +1,337 @@
+import * as t from 'io-ts';
+
+import { POSITIVE_INTEGER_FROM_STRING } from './util';
+
+const FLOW_REF_DIRECTION = t.keyof({
+  source: null,
+  destination: null,
+});
+
+const FLOW_LIST = t.keyof({
+  pending: null,
+  all: null,
+  search: null,
+});
+
+export type FlowList = t.TypeOf<typeof FLOW_LIST>;
+
+const FLOW_OBJECT = t.intersection([
+  t.type({
+    objectID: t.number,
+    refDirection: FLOW_REF_DIRECTION,
+    objectType: t.string,
+  }),
+  t.partial({
+    flowID: t.number,
+    versionID: t.number,
+    behavior: t.union([t.string, t.null]),
+    objectDetail: t.union([t.string, t.null]),
+    createdAt: t.string,
+    updatedAt: t.string,
+  }),
+]);
+
+export type FlowObject = t.TypeOf<typeof FLOW_OBJECT>;
+
+const FLOW_CATEGORY = t.intersection([
+  t.type({
+    name: t.string,
+    group: t.string,
+  }),
+  t.partial({
+    id: t.number,
+    createdAt: t.string,
+    updatedAt: t.string,
+    description: t.union([t.string, t.null]),
+    parentID: t.union([t.number, t.null]),
+    code: t.union([t.string, t.null]),
+    includeTotals: t.union([t.boolean, t.null]),
+    categoryRef: t.type({
+      objectID: t.number,
+      versionID: t.number,
+      objectType: t.string,
+      categoryID: t.number,
+      createdAt: t.string,
+      updatedAt: t.string,
+    }),
+  }),
+]);
+export type FlowCategory = t.TypeOf<typeof FLOW_CATEGORY>;
+
+const CREATED_BY_OR_LAST_UPDATED_BY = t.type({
+  name: t.string,
+});
+
+/** Delete when finishing off REST flow endpoint (Maybe Matyas needs it) */
```

Comment:
(Since this code is old, in the end we need it for add/edit flow, in my latest commits in other branches I reviewed this codec because it was not well defined either)

So we let it like this, in other tickets it's addressed

Reply:
Can we at least not mention former developers? I'd just delete entire comment